vrrpd/vrrp.c

```diff
@@ -1088,6 +1103,27 @@ static int vrrp_socket(struct vrrp_router *r)
 		goto done;
 	}
 
+	/*
+	 * Bind Tx socket to macvlan device - necessary for VRF support,
+	 * otherwise the kernel will select the vrf device
+	 */
+	if (r->vr->ifp->vrf_id != VRF_DEFAULT) {
+		frr_with_privs (&vrrp_privs) {
+			ret = setsockopt(r->sock_tx, SOL_SOCKET,
+					 SO_BINDTODEVICE, r->mvl_ifp->name,
+					 strlen(r->mvl_ifp->name));
+		}
+
+		if (ret < 0) {
+			zlog_warn(
+				VRRP_LOGPFX VRRP_LOGPFX_VRID VRRP_LOGPFX_FAM
+				"Failed to bind Tx socket to macvlan device'%s'",
```

Comment:
shouldn't be whitespace after device? `device '%'`.

Reply:
Right you are